crates/datasources/src/cassandra/mod.rs

```diff
@@ -0,0 +1,167 @@
+mod builder;
+mod errors;
+mod exec;
+use async_stream::stream;
+use async_trait::async_trait;
+use datafusion::arrow::array::{
+    ArrayRef, Float32Builder, Float64Builder, Int16Builder, Int32Builder, Int8Builder,
+    StringBuilder, TimestampMillisecondBuilder,
+};
+use datafusion::arrow::datatypes::{
+    DataType, Field, Fields, Schema as ArrowSchema, SchemaRef as ArrowSchemaRef, TimeUnit,
+};
+
+use datafusion::arrow::record_batch::RecordBatch;
+use datafusion::datasource::TableProvider;
+use datafusion::error::{DataFusionError, Result as DatafusionResult};
+use datafusion::execution::context::SessionState;
+use datafusion::execution::TaskContext;
+use datafusion::logical_expr::{Expr, TableProviderFilterPushDown, TableType};
+use datafusion::physical_expr::PhysicalSortExpr;
+use datafusion::physical_plan::metrics::{ExecutionPlanMetricsSet, MetricsSet};
+use datafusion::physical_plan::{
+    DisplayAs, DisplayFormatType, ExecutionPlan, Partitioning, RecordBatchStream,
+    SendableRecordBatchStream, Statistics,
+};
+use datafusion_ext::metrics::DataSourceMetricsStreamAdapter;
+pub use errors::*;
+use futures::Stream;
+use futures::StreamExt;
+use scylla::frame::response::result::{ColumnType, CqlValue, Row};
+
+use scylla::transport::session::Session;
+use scylla::SessionBuilder;
+use std::any::Any;
+
+use std::fmt;
+use std::pin::Pin;
+use std::sync::Arc;
+use std::task::{Context, Poll};
+
+use self::exec::CassandraExec;
+
+struct CassandraAccess {
+    session: Session,
+}
+
+fn try_convert_dtype(ty: &ColumnType) -> Result<DataType> {
+    Ok(match ty {
+        ColumnType::Ascii => DataType::Utf8,
+        ColumnType::Date => DataType::Date64,
+        ColumnType::Double => DataType::Float64,
+        ColumnType::Duration => DataType::Duration(TimeUnit::Nanosecond),
+        ColumnType::Float => DataType::Float32,
+        ColumnType::Int => DataType::Int32,
+        ColumnType::Text => DataType::Utf8,
+        ColumnType::Timestamp => DataType::Timestamp(TimeUnit::Millisecond, None),
+        ColumnType::SmallInt => DataType::Int16,
+        ColumnType::TinyInt => DataType::Int8,
+        ColumnType::Uuid => DataType::Utf8,
```

Comment:
my gut feeling is that this would be binary rather than a hex string, but I could be wrong. It also looks like that have a UUIDv1 type that they call `timeuuid` and that should be handled the same.

Reply:
We have generally been casting uuid's to strings.